src/uct/tcp/tcp_ep.c

```diff
@@ -258,6 +240,13 @@ static UCS_CLASS_INIT_FUNC(uct_tcp_ep_t, uct_tcp_iface_t *iface, int fd,
     ucs_queue_head_init(&self->pending_q);
     ucs_queue_head_init(&self->put_comp_q);
 
+    if (dest_addr != NULL) {
+        memcpy(&self->peer_addr[0], dest_addr, iface->config.sockaddr_len);
```

Comment:
seems that we assume source and dest address families are the same?
so need to update tcp_iface_is_reachable to check it, to prevent ipv4<->ipv6 connection and wrong usage of iface->config.sockaddr_len